### PR TITLE
Feature/toilet overview paging

### PR DESCRIPTION
**Pull request by Quillraven**
Added paging to the toilet overview page and also changed the root element on `ToiletOverviewView` component from `Container` to `div` to use the entire browser width instead of just like 50% of the width.

There are again some things that I don't like but I guess there is no better solution? These are the things:

- Axios cannot pass a body to a GET request and therefore the `getNearbyToilets` parameters have to be passed as request parameter. 
This is not ideal because the URL gets long and can potentially break if the toilet ids that need to be excluded exceed the 2000 characters limit.
The only way to change that is to make it a non-GET call which also does not make sense to me.
- The second downside of a request parameter is that it does not support `Array` as a type. Therefore, the ids have to be passed as string with a delimeter. Frontend has to convert from array to string and backend has to convert from string to collection.
- `ToiletOverviewView.tsx` contains two functions named `loadToilets` and `onToiletsScroll`. React complains that those functions are missing dependencies of `useEffect` which I understand why **but** when you add them, you get an infinite loop because every time the component renders, those functions get recreated and therefore change --> endless loop.
I thought the solution is `useCallback` but that also requires dependencies and again it will end up in an endless loop when you add them.
I have no idea how to properly fix that. For now I simply deactivated the lint warning on `useEffect` and to me it is working just fine. But still, the warning is there for a reason and there must be a proper solution.
- Last, I did not find a good solution to get a `Toilet` object out of MongoDB with just one additional `distance` field. The reason I need that is because we now have a custom `NearQuery` to support paging. This is done by filtering consecutive calls via a **minDistance**. MongoDB is treating minDistance inclusively and that's why you also have to exclude certain toilet ids in the query.
This is workinng well but I need both information for the overview page - the toilet and the distance info.
The only solution that I found for now is to define a new class containing the necessary information which is a combination of `Toilet` and the additional `distance` field.
I tried adding the distance to the `Toilet` and annotate it with `@Transient` but then the distance information is also not stored when using the `NearQuery`.

To summarize it: everything is working (from what I can tell) but it would be great if there are better solutions for some approaches ;)

### CODE REVIEW

**Comment by Quillraven** [comment]
I read a few articles the last weeks and seems to be the way to go with what we are using.

The only thing left is the `ToiletOverviewView.tsx` ... I try to figure that out over the weekend ;)
